### PR TITLE
Detect language from browser accept languages

### PR DESCRIPTION
**Pull request by mauserrifle**

### CODE REVIEW

File: classes/Translator.php [hunk 1]
```diff
@@ -136,6 +136,23 @@ public function loadLocaleFromRequest()
...
 
+    /**
+     * Sets the locale based on the browser accept languages
+     * @return bool
+     */
+    public function loadLocaleFromRequestLanguages()
+    {
+        $locales = Locale::listAvailable();
+
+        foreach(Request::getLanguages() as $request_lang) {
```
Comment (LukeTowers) [inline classes/Translator.php]: add spaces after the foreach and if please, `$request_lang` to `$requestLang` please. Thanks!

**Comment by LukeTowers** [comment]
@mauserrifle Awesome, thanks! Just waiting on a couple of minor changes.

**Comment by LukeTowers** [comment]
Awesome @mauserrifle! If you can get another person to test and verify that it works then I'll merge. Perhaps @datune or @vannut?

**Comment by datune** [comment]
@mauserrifle @LukeTowers @vannut 

I tend to disagree with this approach. It makes no sense to couple the detection of the locale sent from the browser based on whether or not I want the default locale prefixed in the URL. 

Detecting browser language preferences should work indepenently of whether or not I want the default locale prefixed. 

I'd much rather see an additional config setting allowing us to turn this feature on or off.

I took a stab at it, but it's not working with prefixDefaultLocale set to off. I'm not sure why. Maybe @vannut can come to the rescue?

```php
<?php namespace RainLab\Translate\Classes;

use RainLab\Translate\Classes\Translator;
use Closure;
use Config;

class LocaleMiddleware
{
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @return mixed
     */
    public function handle($request, Closure $next)
    {
        $translator = Translator::instance();
        $translator->isConfigured();

        if (!$translator->loadLocaleFromRequest()) {
            if (Config::get('rainlab.translate::prefixDefaultLocale')) {
                if (!$translator->loadLocaleFromSession()) {
                    $this->loadLocaleFromRequestLanguages($translator);
                }
            }
            else if (!$this->loadLocaleFromRequestLanguages($translator)) {
                $translator->setLocale($translator->getDefaultLocale());
            }
        }

        return $next($request);
    }

    /*
     * Set Locale based on the accepted Request (usually browser) languages
     *
     * @return bool
     */
    protected function loadLocaleFromRequestLanguages($translator)
    {
        if (Config::get('rainlab.translate::setLocaleFromRequestLanguages')) {
            return $translator->loadLocaleFromRequestLanguages();
        }

        return false;
    }
}

```

**Comment by vannut** [comment]
> ...we would be able to add aliases to our defined language codes in the backend...

Maybe that's not a bad idea: adding aliasses to a certain 'cms-locale' with a fallback to the default.

**Comment by Samuell1** [comment]
This plugin mostly does what you need guys:

https://github.com/Excodus/oc-translate-extended/blob/master/routes.php#L42

**Comment by mjauvin** [comment]
Closing as it has been over a month since any activity on this occurred and we are trying to figure out what issues are still relevant. If this is still something that you would like to see through to fruition please respond and we can get the ball rolling.

**Pull request closed without merging**